models/user.js

```diff
@@ -0,0 +1,42 @@
+const bcrypt = require('bcrypt')
+const mongoose = require('mongoose')
+
+const UserSchema = new mongoose.Schema({
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+        match: [/^.+@(?:.+\.)?ua\.edu$/, 'Email must be a valid ua.edu address']
```

Comment:
I'm not sure about this regular expression. It seems like it could support something like `myemail@anyrandomstring.ua.edu` which is not a valid UA email. I think we should strictly enforce either `@crimson.ua.edu` or `@ua.edu` with something like `/^.+@(?:crimson\.)?ua\.edu$/`